remoteworkitem/remoteworkitem.go

```diff
@@ -3,54 +3,74 @@ package remoteworkitem
-	GithubAssignee    = "assignee.login"
+	// The keys in the flattened response JSON of a typical Github issue.
+	GithubTitle         = "title"
+	GithubDescription   = "body"
+	GithubState         = "state"
+	GithubID            = "url"
+	GithubCreator       = "user.login"
+	GithubAssigneeLogin = "assignee.login"
+	GithubAssigneeURL   = "assignee.url"
 
 	// The keys in the flattened response JSON of a typical Jira issue.
+	JiraTitle         = "fields.summary"
+	JiraBody          = "fields.description"
+	JiraState         = "fields.status.name"
+	JiraID            = "self"
+	JiraCreator       = "fields.creator.key"
+	JiraAssigneeLogin = "fields.assignee.key"
+	JiraAssigneeURL   = "fields.assignee.self"
+)
 
-	JiraTitle    = "fields.summary"
-	JiraBody     = "fields.description"
-	JiraState    = "fields.status.name"
-	JiraID       = "self"
-	JiraCreator  = "fields.creator.key"
-	JiraAssignee = "fields.assignee"
+// RemoteWorkItem a temporary structure that holds the relevant field values retrieved from a remote work item
+type RemoteWorkItem struct {
+	// The field values, according to the field type
+	Fields map[string]interface{}
+	// unique id per installation
+	ID string
+	// Name of the type of this work item
+	Type string
+}
 
-	ProviderGithub = "github"
-	ProviderJira   = "jira"
+const (
+	remoteTitle               = workitem.SystemTitle
+	remoteDescription         = workitem.SystemDescription
+	remoteState               = workitem.SystemState
+	remoteItemID              = workitem.SystemRemoteItemID
+	remoteCreator             = workitem.SystemCreator
+	remoteAssigneeLogins      = "system.assignee.login"
+	remoteAssigneeProfileURLs = "system.assignee.profile_url"
 )
 
-// WorkItemKeyMaps relate remote attribute keys to internal representation
-var WorkItemKeyMaps = map[string]WorkItemMap{
+// RemoteWorkItemKeyMaps relate remote attribute keys to internal representation
+var RemoteWorkItemKeyMaps = map[string]RemoteWorkItemMap{
 	ProviderGithub: {
-		AttributeMapper{AttributeExpression(GithubTitle), StringConverter{}}:                                             workitem.SystemTitle,
-		AttributeMapper{AttributeExpression(GithubDescription), MarkupConverter{markup: rendering.SystemMarkupMarkdown}}: workitem.SystemDescription,
-		AttributeMapper{AttributeExpression(GithubState), GithubStateConverter{}}:                                        workitem.SystemState,
-		AttributeMapper{AttributeExpression(GithubID), StringConverter{}}:                                                workitem.SystemRemoteItemID,
-		AttributeMapper{AttributeExpression(GithubCreator), StringConverter{}}:                                           workitem.SystemCreator,
-		AttributeMapper{AttributeExpression(GithubAssignee), ListStringConverter{}}:                                      workitem.SystemAssignees,
+		AttributeMapper{AttributeExpression(GithubTitle), StringConverter{}}:                                             remoteTitle,
```

Comment:
@xcoulon You could omit the `AttributeMapper` typing here. Go will infer the type.  

```
ProviderGithub: {
		{AttributeExpression(GithubTitle), StringConverter{}}:                                             workitem.SystemTitle,
		{
...
```